### DISCUSSION TITLE
Avalanche unable to use cuda device.

### DISCUSSION THREAD

**Discussion by jodie-kang** [ContinualAI/avalanche#1582] [Q&A]
🐛 **Describe the bug**
UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11050). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.) 

🐜 **To Reproduce**

Installation Process: 

- conda create -n avalanche-env python=3.8 
- conda activate avalanche-env 
- pip install -i https://pypi.tuna.tsinghua.edu.cn/simple avalanche-lib  

 Observations: Automatic installation of torch-2.1.0 

Attempt 1: Manual installation of the GPU version of PyTorch: 
conda install pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.3 -c pytorch 

Encountered an error: TypeError: BaseSGDTemplate.init() missing 2 required positional arguments: 'model' and 'optimizer' 

Observation: The above command led to automatic uninstallation of torch-2.1.0 

Attempt 2: pip install typing-extensions==4.4.0, but the problem persists 

Attempt 3: Created a conda environment with Python=3.10, but the issue remains unresolved.

🐝 **Expected behavior**
Due to certain reasons, I lack permission to change the CUDA version on the server. Are there any alternative methods to address the above issue?

🐞 **Screenshots**
![image](https://github.com/ContinualAI/avalanche/assets/88081081/ef2e7a88-0358-48b3-b679-61e7168be317)

🦋 **Additional context**
NVIDIA-SMI 495.46 Driver Version: 495.46 CUDA Version: 11.5 
![image](https://github.com/ContinualAI/avalanche/assets/88081081/82202bf6-46c7-4f09-ac4e-7fbaf15da6cf)

**Comment by jodie-kang** [comment]
I first installed torch 1.13.1, where torch.cuda.is_available() returned True. 
However, after installing avalanche-lib, torch 1.13.1 was automatically uninstalled and torch 2.1.0 was forcefully installed, resulting in torch.cuda.is_available() returning False.
![image](https://github.com/ContinualAI/avalanche/assets/88081081/6adcd7b8-9d7e-4948-92ed-3efcc8c0d09f)

**Comment by jodie-kang** [comment]
avalanche-lib：0.4.0
![image](https://github.com/ContinualAI/avalanche/assets/88081081/da5641ac-ca1f-48a0-b5b3-c960a43d76ff)

**Comment by AntonioCarta** [comment]
looks like an environment issue due to installing pytorch with conda and avalanche with pip.
Try to install both with pip. If you still have the issue, try:
```
conda install typing_extensions==4.4.0
pip install typing_extensions==4.4.0
```
